packages/client/src/guards/admin.guard.tsx

```diff
@@ -0,0 +1,24 @@
+import { FC } from 'react';
+import { useAuth } from '@context/auth.context';
+import { Paths } from '@constants/paths';
+import { Outlet, useNavigate } from 'react-router-dom';
+import { useProject } from '@context/project.context';
+
+export const AdminGuard: FC = () => {
+  const { token, decoded_token, initialized } = useAuth();
+  const { project } = useProject();
+  const navigate = useNavigate();
+  const projectId = project?.id;
+  const loginUrl = `${import.meta.env.VITE_AUTH_CLIENT}?projectId=${projectId}&redirectUrl=${encodeURIComponent(window.location.origin + Paths.AUTH_CALLBACK)}`;
+
+  if (initialized && !token) {
+    window.location.replace(loginUrl);
+    return null;
+  }
+
+  if (initialized && decoded_token?.role !== 1) {
+    navigate(Paths.PERMISSION_REQUIRED);
+  }
```

Comment:
You should use useEffect around this logic. So that way it re-runs when props change